### PR TITLE
cmd: Add support for AWS Profile/Config

### LINKED ISSUE #48: Support AWS profiles/config
Don't insist on an AWS key/secret-key, ideally use profiles as the standard AWS SDK supports them.

Accepting a session-token parameter might be a vaguely acceptable mid-way step for your code...

### PR DESCRIPTION
**Pull request by xescugc**
And also made it more TF/AWS compatible by adopting the same flags and ENV variables
so if you know how to configure any of the others it'll work by default

Closes #48

### CODE REVIEW

File: cmd/aws.go [hunk 1]
```diff
@@ -98,10 +113,36 @@ func init() {
 	awsCmd.AddCommand(awsResourcesCmd)
 
 	// Required flags
-	awsCmd.Flags().String("access-key", "", "Access Key (required)")
-	awsCmd.Flags().String("secret-key", "", "Secret Key (required)")
-	awsCmd.Flags().String("region", "", "Region to search in, for now * it's not supported (required)")
+	awsCmd.Flags().String("aws-access-key", "", "Access Key (required)")
+	awsCmd.Flags().String("aws-secret-access-key", "", "Secret Key (required)")
+	awsCmd.Flags().String("aws-default-region", "", "Region to search in, for now * it's not supported (required)")
```
Comment (xlr-8) [inline cmd/aws.go:118]: is not supported*

File: cmd/aws.go [hunk 2]
```diff
@@ -98,10 +113,36 @@ func init() {
...
 	awsCmd.Flags().StringSliceVarP(&tags, "tags", "t", []string{}, "List of tags to filter with format 'NAME:VALUE'")
 }
+
+// loadAWSCredentials will first read from ENV and if AccessKey and SecretAccessKey are not found (both of them)
+// will fallback to the SharedCredentials with the profile
+func loadAWSCredentials() error {
+	creds := credentials.NewCredentials(&credentials.ChainProvider{
+		Providers: []credentials.Provider{
+			&credentials.EnvProvider{},
+			&credentials.SharedCredentialsProvider{Filename: viper.GetString("aws-shared-credentials-file"), Profile: viper.GetString("aws-profile")},
```
Comment (xlr-8) [inline cmd/aws.go:132]: Shouldn't define a const for this strings to avoid typos?

File: cmd/aws.go [hunk 3]
```diff
@@ -98,10 +113,36 @@ func init() {
...
+		},
+	})
+
+	value, err := creds.Get()
+	if err != nil {
+		if awsE, ok := err.(awserr.Error); ok && awsE.Code() == "NoCredentialProviders" {
+			return nil
```
Comment (xlr-8) [inline cmd/aws.go:139]: Not sure why this error is ignored? Don't you end up without any credential in this situation?
Or it's just that the SharedCredentialProvider that fails? If so, it's still confusing as you commented: if ENV fails, fallback on Shared Credential Profile - so at this point it seems like there are no valid credentials found

File: cmd/aws.go (hunk 2 above)
Comment (xescugc) [reply cmd/aws.go:132]: Well I could, but then I should also define those constants for all the flags we have, as this is basically the flag name.

File: cmd/aws.go (hunk 3 above)
Comment (xescugc) [reply cmd/aws.go:139]: I added a comment to add more context to it (I planned to add it as it was also confusing for me haha).

> // The NoCredentialProviders is an error returned by Get to identify that none
// of the Providers (credentials.EnvProvider and credentials.SharedCredentialsProvider)
// did find any information.
// So we escape it means nothing was found by AWS

Basically it's an specific error saying that AWS did not find any credentials with the information provided, which we do not care as this is just a step to fill the credentials and after this step we do a validation of the required flags.

So we ignore this errors as it's confusing in this context and not needed as we only need the values if there are any defined.

**Comment by xescugc** [comment]
Done/Commented

File: cmd/aws.go (hunk 2 above)
Comment (xlr-8) [reply cmd/aws.go:132]: If they are re-used several times, sounds fine to me :smile:

File: cmd/aws.go (hunk 3 above)
Comment (xlr-8) [reply cmd/aws.go:143]: Hum, can the code run when going through the nil, or it simply will fail later on?
If it's gonna fail later on, I'd rather return a proper error here

Comment (xescugc) [reply cmd/aws.go:143]: It'll fail later on as those values will be empty and we have a validation that the required ones are not empty.

Why return a proper error here? Which is the benefit? What is saying is "I did not find anything" why do we care? We'll always ignore it as it's not required for the user to use profile or env, they can still define things via flags which is out of this specific logic.

Comment (xlr-8) [reply cmd/aws.go:143]: OK, then it was my misunderstanding - if the code can still run later on - good for me

**Comment by xescugc** [comment]
Pushed new changes to not make the new function override flags if defined.

**Comment by xescugc** [comment]
Rebased

**Review by xlr-8 — APPROVED** [review]

**Pull request merged**